pkg/activator/podip_watcher.go

```diff
@@ -0,0 +1,258 @@
+package activator
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging/logkey"
+	"knative.dev/serving/pkg/apis/networking"
+	"knative.dev/serving/pkg/apis/serving"
+	"knative.dev/serving/pkg/network"
+	"knative.dev/serving/pkg/network/prober"
+	"knative.dev/serving/pkg/queue"
+	"knative.dev/serving/pkg/reconciler"
+	"knative.dev/serving/pkg/resources"
+)
+
+const probePortName string = "http"
```

Comment:
What about H2C?

Reply:
Right now we only probe via HTTP in activator, IMO this makes sense as our qeue health check is independent of the underlying app proto and queue-proxy can implement an h2c check if that is what the app is using.